src/pages/Intercepted.js

```diff
@@ -31,28 +31,26 @@ class Intercepted extends React.Component {
             let timestamp = new Date().getTime();
             let timePassed = timestamp - this.state.timestamp;
 
-            isCurrentTab().then(isCurrentTab => {
-                if (!isCurrentTab) timePassed = 0;
+            if (!document.hasFocus()) timePassed = 0;
 
-                let timeLeft = this.state.timeLeft - timePassed;
-                
-                if (timeLeft <= 0) clearInterval(this.state.timer)
-    
-                // update time spent learning on website
-                getFromStorage('timeSpentLearning').then(res => {
-                    let timeSpentLearning = res.timeSpentLearning || {};
-                    let site = this.getExerciseSite();
-    
-                    if (!site) return; // not found, do not update.
-    
-                    let newExerciseTimeSpent = timeSpentLearning[site.name]
-                                                 + timePassed || timePassed;
-                    timeSpentLearning[site.name] = newExerciseTimeSpent;
-                    return setInStorage({ timeSpentLearning });
-                });
-    
-                this.setState({ timeLeft, timestamp });
+            let timeLeft = this.state.timeLeft - timePassed;
+            
+            if (timeLeft <= 0) clearInterval(this.state.timer)
+
+            // update time spent learning on website
+            getFromStorage('timeSpentLearning').then(res => {
+                let timeSpentLearning = res.timeSpentLearning || {};
+                let site = this.getExerciseSite();
+
+                if (!site) return; // not found, do not update.
+
+                let newExerciseTimeSpent = timeSpentLearning[site.name]
+                                                + timePassed || timePassed;
```

Comment:
does time passed continue counting even if the progress bar is finished?
my impression is that if i set a timer to 30seconds, and i do three minutes, only 30 seconds are added anyway.

Reply:
No time passed does not continue counting, this is intentional. New UI/UX would have to be added in order to make this clear to the user. You could add a feature request if you'd like.